### ISSUE TITLE
sql: CREATE TABLE LIKE does not copy predicates of partial indexes

### ISSUE THREAD

**Issue by mgartner** [cockroachdb/cockroach#51849]
Notice that table `u` below is missing the `WHERE b > 1:::INT8` predicate in its partial index.

```
root@127.0.0.1:49276/defaultdb> create table t (a int, b int, index (a) where b > 1);
CREATE TABLE

root@127.0.0.1:49276/defaultdb> show create table t;
  table_name |               create_statement
-------------+------------------------------------------------
  t          | CREATE TABLE t (
             |     a INT8 NULL,
             |     b INT8 NULL,
             |     INDEX t_a_idx (a ASC) WHERE b > 1:::INT8,
             |     FAMILY "primary" (a, b, rowid)
             | )
(1 row)

root@127.0.0.1:49276/defaultdb> create table u (like t including indexes);
CREATE TABLE

root@127.0.0.1:49276/defaultdb> show create table u;
  table_name |          create_statement
-------------+-------------------------------------
  u          | CREATE TABLE u (
             |     a INT8 NULL,
             |     b INT8 NULL,
             |     INDEX t_a_idx (a ASC),
             |     FAMILY "primary" (a, b, rowid)
             | )
(1 row)
```

**Comment by jordanlewis** [comment]
Nice find! We should consider a randomized test for `CREATE TABLE LIKE` that creates a random table, runs `CREATE TABLE LIKE` on it, and verifies that the table definition is the same before and after.

**Comment by mgartner** [comment]
Are there examples of randomized tests I can use as a starting point?

**Comment by jordanlewis** [comment]
We already have something called `RandCreateTable` that you could reuse. Check out `pkg/sql/tests/random_schema_test.go`, which doesn't seem to contain that much, but you could use this as a decent starting point.

You might need to enhance `RandCreateTable` to include partial indexes too, to make this work :)

**Comment by rohany** [comment]
Adding partial indexes to RandCreateTable would get you inclusion in SQLSmith for free

**Comment by mgartner** [comment]
I've extended support to `RandCreateTable` to build random partial indexes, but partial indexes are gated by the `experimental_partial_indexes` session setting, so tests like `TestCreateRandomSchema` fail.

Are there examples where we use gated features in random tests or sqlsmith? Should this wait until that session flag is removed? I'd like to get this `RandCreateTable` change in sooner rather than later so that we can start to catch any potential bugs ASAP. Any ideas @rohany @jordanlewis ?

**Comment by rohany** [comment]
If you search for uses of `experimental_enable_enums` you'll find some. I think it's a bit easier in the cases that i've included it because enums aren't in some of these general methods. Perhaps you could introduce a a new enum argument to `RandCreateTable` about what features to include, but the plumbing here might be too much. I think it should be easy to add partial indexes to the SQLSmith runs as part of some of the seed setups without putting partial indexes in methods like `RandCreateTable`.

**Comment by mgartner** [comment]
I created a new ticket to track adding partial indexes to randomized tests and writing a randomized CREATE TABLE LIKE test. #51849
